tools/sambamba/Sambamba_markdup.xml

```diff
@@ -1,51 +1,48 @@
-<tool id="sambamba_markdup" name="MarkDup" version="@WRAPPER_VERSION@">
-    <description>Finds duplicate reads in BAM files</description>
+<tool id="sambamba_markdup" name="Sambamba markdup" version="@TOOL_VERSION@+galaxy@SUFFIX_VERSION@" profile="23.2" license="MIT">
+    <description>Finds and marks duplicate reads in BAM files</description>
     <macros>
-        <import>Sambamba_macros.xml</import>
+        <import>macros.xml</import>
     </macros>
     <expand macro="requirements"/>
-    <expand macro="stdio"/>
-
-    <command><![CDATA[
-        @EXECUTABLE markdup 
-               $Remove_duplicates
-            -l $Compression_level
-               '${input_file1}'
+    <command detect_errors="exit_code"><![CDATA[
+        sambamba markdup 
+               $remove_duplicates
+            -l $compression_level
+               '${input_bam}'
                '${output}'
+            --nthreads \${GALAXY_SLOTS:-4}
+            --tmpdir '\${TMPDIR:-.}'
     ]]></command>
-
     <inputs>
-        <expand macro="inputSetup1"/>
-        <expand macro="compression_level"/>
-
-        <param name="Remove_duplicates"
-               type="boolean"
-               truevalue="-r"
-               falsevalue=""
-               argument="-r"
-               checked="false"
-               label="Remove duplicates instead of just marking them"/>
+        <!-- For deduplication the input bam needs to be coordinate sorted, hence the input file is expected to be coordinate sorted -->
+        <param name="input_bam" type="data" format="bam" label="Input BAM file"/>
+        <param argument="--compression-level" type="integer" value="5" min="0" max="9" label="Level of compression for merged BAM file, number from 0 to 9"/>
+        <param argument="--remove-duplicates" type="boolean" truevalue="-r" falsevalue="" label="Remove duplicates instead of just marking them"/>
     </inputs>
-
     <outputs>
         <data name="output" format="bam"/>
     </outputs>
-
     <tests>
-        <test>
-            <param name="input_file1" value="2.bam" ftype="bam"/>
-            <param name="Compression_level" value="2"/>
-            <param name="Remove_duplicates" value="-r"/>
-
-            <output name="output" file="MarkDup.bam" compare="sim_size" delta="434"/>
+        <!-- sambamba markdup without dropped reads -->
+        <test expect_num_outputs="1">
+            <param name="input_bam" value="2.bam" ftype="bam"/>
+            <param name="compression_level" value="5"/>
+            <param name="remove_duplicates" value="false"/>
+            <output name="output" file="2.markdup.bam" compare="sim_size" delta="434"/>
```

Comment:
bam files can be diffed ... in the special case of BAM Galaxy will uncrompress BAM and compare the SAM